src/native/libs/System.Security.Cryptography.Native.Browser/pal_crypto_webworker.js

```diff
@@ -0,0 +1,47 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+const CryptoWebWorkerLib = {
+    $CRYPTOWEBWORKER: {
+        call_digest: function (hash, input_buffer, input_len, output_buffer, output_len) {
+            if (this.can_call_digest() !== 1) {
+                return 0; // Not supported. Caller should have validated this first.
+            }
+
+            var msg = {
+                func: "digest",
+                type: hash,
+                data: Array.from(Module.HEAPU8.subarray (input_buffer, input_buffer + input_len))
+            };
+
+            var response = globalThis.mono_wasm_crypto.channel.send_msg (JSON.stringify (msg));
+            var digest = JSON.parse (response);
+            if (digest.length > output_len) {
+                console.info("call_digest: about to throw!");
+                throw "DIGEST HASH: Digest length exceeds output length: " + digest.length + " > " + output_len;
+            }
+
+            Module.HEAPU8.set (digest, output_buffer);
+            return 1;
+        },
+        can_call_digest: function () {
+            if (typeof globalThis.mono_wasm_crypto === "undefined" ||
+                typeof globalThis.mono_wasm_crypto.channel === "undefined" ||
+                typeof globalThis.mono_wasm_crypto.worker === "undefined" ||
+                typeof SharedArrayBuffer === "undefined") {
+                return 0;
+            }
```

Comment:
`call_digest` has the comment that `crypto.subtle.[...]` is not available in all execution environments.  Should `can_call_digest` check for those to say false?

Reply:
I've updated the logic such that the correct check is here - https://github.com/dotnet/runtime/blob/0f859511076c97e5cd4b89789616b69286819215/src/mono/wasm/runtime/startup.ts#L54-L55

That is the source of truth, and in this method we'll rely on `globalThis.mono_wasm_crypto.*` only being initialized if the APIs can be used.